internal/storage/payments.go

```diff
@@ -64,6 +64,7 @@ type paymentAdjustment struct {
 
 func (s *store) PaymentsUpsert(ctx context.Context, payments []models.Payment) error {
 	paymentsToInsert := make([]payment, 0, len(payments))
+	paymentsToInsertSeen := make(map[models.PaymentID]int)
```

Comment:
Let's avoid modifying this file and try to get everything to work with the `REFUNDED` flow.